### PR TITLE
Make build scope aware

### PR DESCRIPTION
**Pull request by dkarnutsch**
This requires the CronJob and the Job to be annotated with comet-dxp.com/content-scope (done with https://gitlab.vivid-planet.com/comet/charts/-/merge_requests/49/)

### CODE REVIEW

**Comment by dkarnutsch** [comment]
The annotation is not mandatory. `ContentScope` can be undefined for `isContentScopeAllowed`.

I had it before, that isContentScopeAllowed is optional, but I didn't like the API. I'll give it another try.

**Comment by johnnyomair** [comment]
You could add a default implementation `() => true` when no explicit method is defined

**Comment by dkarnutsch** [comment]
As suggested by @johnnyomair, I've changed it to use the `ContentScopeModule`.  Add @nsams as reviewer, because some changes were needed (Creating ContentScopeService to use `canAccessScope` outside of the module).

**Comment by dkarnutsch** [comment]
Changed the base to next as this is breaking (`ContentScopeModule` needs to be added to the AppModule)

File: packages/api/cms-api/src/builds/changes-checker.console.ts [hunk 1]
```diff
@@ -22,7 +19,7 @@ export class ChangesCheckerConsole {
 
         if (await this.buildsService.hasChangesSinceLastBuild()) {
             console.log("Changes detected, starting build...");
-            await this.buildsService.createBuilds("changesDetected");
+            await this.buildsService.dangerouslyCreateBuildsWithoutUserCheck("changesDetected");
```
Comment (johnnyomair) [inline packages/api/cms-api/src/builds/changes-checker.console.ts:22]: We're still planning to only start selected build based on the changes, am I right?

**Review by johnnyomair — APPROVED** [review]

File: packages/api/cms-api/src/builds/changes-checker.console.ts (hunk 1 above)
Comment (dkarnutsch) [reply packages/api/cms-api/src/builds/changes-checker.console.ts:22]: absolutely

File: packages/api/cms-api/src/content-scope/content-scope.module.ts [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
 
 import { CurrentUser } from "../auth/dto/current-user";
 import { CAN_ACCESS_SCOPE } from "./conent-scope.constants";
+import { ContentScopeService } from "./content-scope.service";
 import { ScopeGuard } from "./scope.guard";
 
+export type CanAccessScope = (requestScope: ContentScope, user: CurrentUser) => boolean;
+
 interface ContentScopeModuleOptions {
     canAccessScope: (requestScope: ContentScope, user: CurrentUser) => boolean;
```
Comment (nsams) [inline packages/api/cms-api/src/content-scope/content-scope.module.ts:13]:
```suggestion
    canAccessScope: CanAccessScope;
```

File: packages/api/cms-api/src/builds/builds.service.ts [hunk 3]
```diff
@@ -70,15 +76,26 @@ export class BuildsService {
...
+     * Should only be used internally
+     */
+    async dangerouslyCreateBuildsWithoutUserCheck(trigger: string): Promise<boolean> {
+        const builderCronJobs = await this.kubernetesService.getAllCronJobs(
+            `${BUILDER_LABEL} = true, ${INSTANCE_LABEL} = ${this.kubernetesService.helmRelease}`,
+        );
+        return this.createBuilds(trigger, builderCronJobs);
+    }
+
+    async createBuildsWithUserCheck(trigger: string, builderCronJobs: V1CronJob[]): Promise<boolean> {
```
Comment (nsams) [inline packages/api/cms-api/src/builds/builds.service.ts:89]: I don't understand this: what about the user is checked here? 

And why is this user check needed at all? (in addition to the .canAccessScope check in createBuilds)

Comment (dkarnutsch) [reply packages/api/cms-api/src/builds/builds.service.ts]: @nsams you are absolutely right. The check, which is not there, is not needed. I've fixed the name.

**Review by johnnyomair — DISMISSED** [review]

**Review by johnnyomair — APPROVED** [review]

File: packages/api/cms-api/src/builds/builds.service.ts [hunk 4]
```diff
@@ -70,15 +76,22 @@ export class BuildsService {
         return true;
     }
 
-    async getBuilds(options?: { limit?: number | undefined }): Promise<BuildObject[]> {
+    /**
+     * Should only be used internally
+     */
+    async dangerouslyCreateBuildsWithoutUserCheck(trigger: string): Promise<boolean> {
```
Comment (nsams) [inline packages/api/cms-api/src/builds/builds.service.ts:82]: could this be named to what it actually does? eg. createBuildsForAllScopes? (If I understand that correctly)

**Comment by nsams** [comment]
and the admin part for this is coming up in a second PR?

File: packages/api/cms-api/src/builds/builds.service.ts (hunk 4 above)
Comment (dkarnutsch) [reply packages/api/cms-api/src/builds/builds.service.ts]: I agree. Renamed.

**Comment by dkarnutsch** [comment]
> and the admin part for this is coming up in a second PR?

@nsams For now the admin needs no changes. The BuildsTable only returns the builds you are allowed to see and when you want to start new builds, you only see the BuildTemplates your are allowed to see.

**Review by nsams — APPROVED** [review]

**Pull request merged**